src/ontodia/data/rdf/rdfParsAdapter.ts

```diff
@@ -0,0 +1,99 @@
+import { Parsers } from 'rdf-ext';
+import { Dictionary } from '../model';
+import { Stream, Quad, namedNode, literal, blankNode, Node, quad } from 'rdf-data-model';
+
+export class RDFParsAdapter {
+    constructor(private _super: any) { }
+    import(datastream: any): any {
+        let fullString = '';
+
+        let onDataCallbacks: ((quad: Quad) => void)[] = [];
+        let onDataEndCallbacks: ((quad: Quad) => void)[] = [];
+        let onErrorCallbacks: ((error: Error) => void)[] = [];
+
+        let endOfStream = false;
+        let parseError: Error;
+
+        datastream.on('data', (bytes: number[]) => {
+            fullString += this.byteArrayToString(bytes);
+        });
+
+        datastream.on('end', () => {
+            this._super.parse(fullString).then((rdfGraph: any) => {
+                const quads = this.triplesToQuads(rdfGraph.toArray());
+
+                if (onDataCallbacks) {
+                    for (const quad of quads) {
+                        for (const callback of onDataCallbacks) {
+                            callback(quad);
+                        }
+                    }
+                }
+                endOfStream = true;
+                if (onDataEndCallbacks) {
+                    for (const callback of onDataEndCallbacks) {
+                        callback(null);
+                    }
+                }
+            }).catch((error: Error) => {
+                parseError = error;
+                endOfStream = true;
+                if (onErrorCallbacks) {
+                    for (const callback of onErrorCallbacks) {
+                        callback(error);
+                    }
+                }
+            });
+        });
+
+        const on = (event: string, calback: (data?: (Quad | Error)) => void) => {
+            if (event === 'data') {
+                onDataCallbacks.push(calback);
+            } else if (event === 'end') {
+                onDataEndCallbacks.push(calback);
+                if (endOfStream) {
+                    calback(null);
+                }
+            } else if (event === 'error') {
+                onErrorCallbacks.push(calback);
+                if (parseError) {
+                    calback(parseError);
+                }
+            }
+        };
+
+        return {
+            on: on,
+            once: on,
```

Comment:
`once` event API should ["register a listener that is called at most once for a particular event"](https://nodejs.org/api/events.html#events_handling_events_only_once)